### DISCUSSION TITLE
Query parameter list / multiple values : Endpoint works but the Swagger docs not updated as expected

### DISCUSSION THREAD

**Discussion by UbeyOzcan** [fastapi/fastapi#11494] [Questions] [answered]
### First Check

- [X] I added a very descriptive title here.
- [X] I used the GitHub search to find a similar question and didn't find it.
- [X] I searched the FastAPI documentation, with the integrated search.
- [X] I already searched in Google "How to X in FastAPI" and didn't find any information.
- [X] I already read and followed all the tutorial in the docs and didn't find an answer.
- [X] I already checked if it is not related to FastAPI but to [Pydantic](https://github.com/pydantic/pydantic).
- [X] I already checked if it is not related to FastAPI but to [Swagger UI](https://github.com/swagger-api/swagger-ui).
- [X] I already checked if it is not related to FastAPI but to [ReDoc](https://github.com/Redocly/redoc).

### Commit to Help

- [X] I commit to help with one of those options 👆

### Example Code

```python
from typing import Annotated

from fastapi import FastAPI, Query

app = FastAPI()


@app.get("/items/")
async def read_items(q: Annotated[list[str] | None, Query()] = None):
    query_items = {"q": q}
    return query_items
```

### Description

- Open the Browser and open the Endpoint http://localhost:8000/items/?q=foo&q=bar&q=bar&q=bar
- The expected Json output is {"q":["foo","bar","bar","bar"]}
- The actual Json output is {"q":["foo","bar","bar","bar"]} 
- But, the API documentation is not updated : 
![API docs](https://github.com/tiangolo/fastapi/assets/40140140/6d1a66bd-a6c0-4db0-8179-83e5e8c7da41)
- The expected API Documentation should looks like this : 
![API docs correct](https://github.com/tiangolo/fastapi/assets/40140140/0a013670-e53c-481d-9d5a-09792850c3ac)

Anyone has this behaviour as well with the Query parameter list / multiple values tutorial ?

Many thanks in advance !

### Operating System

Windows

### FastAPI Version

0.109.1

### Pydantic Version

2.7.1

### Python Version

3.12.3

**Comment by YuriiMotov** [comment, marked as answer]
It's a well-known issue. Use `SkipJsonSchema[None]` instead of `None` in type annotation:

```
async def read_items(q: Annotated[list[str] | SkipJsonSchema[None], Query()] = None):
```

**Reply by UbeyOzcan** [reply]
It works perfectly thank you for the support ! :)

**Comment by NewSouthMjos** [comment]
Can we add a note to the docs about this issue? It is confusing when implementing straightforward code from the docs leads to different behavior. It took me a while to find this discussion.
